module-4/backend/week_three.md

```diff
@@ -1,19 +1,32 @@
 ## Week Three - Module 4 Recap
-
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
-
-Note: When you're done, submit a PR. 
-
 1. At a high level, what is Node?
+  * Node is a runtime that allows for JavaScript to be run outside of a browser.
 2. What is Express? What is Express similar to in the Ruby world?
+  * Express is a framework and is similar to Rails in the Ruby world.
 3. How do we setup a route when creating an API with Node and Express? Please provide a code snippet.
+  * Inside the app.js file a variable will be declared that points to the routes file.
+    ``` var index = require './routes/index'```
+    This variable will also need to be "used" with in the app.js file as such: `app.use('/', index) `
+    Then inside the routes index file we will have the actually http request verb:
+    ```router.get('/', indexController.index)```
 4. What do we use Knex for?
+  * Knex is used as a query builder and would be similar to an ORM.
 5. How could you organize your code to follow the MVC design pattern in your Quantified Self project?
+  * Model is where queries to the database would be created and a Controller would call those functions within the model to serve the correct info.
 6. How do you execute raw SQL in node?
+  * Using the Knex function raw, raw SQL queries can be written to interact with the database.
 7. What are some advantages to having your front end and back end code live in separate applications? What are some disadvantages?
+  * Advantages would be that the two existing codes can be more flexible and potentially be used in more than one project. The disadvantage is ensuring that they are "hooked" up correctly and if something changes on either end it is not clear how that could potentially affect the other code base.
 
 #### Review  
-
 8. Describe DNS.
+  * Domain Name System: It's how an http request goes about finding the correct information to serve up a response.
```

Comment:
Yes, I'd recommend being prepared to go into more detail for interviews! We will review in class, too😊